lua/bufferline/state.lua

```diff
@@ -52,24 +52,26 @@ local state = {
 }
 
 --- Get the state of the `id`
---- @param id integer the `bufnr`
+--- @param bufnr integer the `bufnr`
 --- @return bufferline.state.data
-function state.get_buffer_data(id)
-  local data = state.buffers_by_id[id]
+function state.get_buffer_data(bufnr)
+  local data = state.data_by_bufnr[bufnr]
 
   if data ~= nil then
     return data
+  elseif bufnr == 0 then
+    bufnr = get_current_buf()
```

Comment:
There was a bug where, by passing in 0 (which is a valid buffer number in most contexts), it would initialize a new data entry at index zero instead of evaluating 0's meaning (i.e. `nvim_get_current_buf`).